### DISCUSSION TITLE
Status report?

### DISCUSSION THREAD

**Discussion by clearoak** [WikidPad/WikidPad#81] [General]
As a mere very long-time user of Wikidpad on Linux, and before that Windows, I am interested to know whether I'll still be able to run it in years to come.   I think someone mentioned that the developers were working on it but there seems to be no information here.   I use it for the most basic things, I suppose, the recording of text, so all the bells and whistles don't interest me.   It's the basic functioning that needs to be solid and reliable, and I'm not certain if this will continue.   So a status report every, erm, year, perhaps, would be most welcome.

**Comment by BjornFJohansson** [comment]
I am using wikidpad every day since 2006 (6786 pages and counting). I am not interested in more features, but only to keep it working. I am not sure we have any developers anymore, it seem like Michael Butcher has perhaps stepped down.

**Reply by DendrocoposMajor** [reply]
It would've been nice to actually know if Michael Butcher has disembarked for perpetuity or if it's a long, very long absence?
Anyway, while I to some extent agree functionality wise, Python 3.n offers some benefits (incl. speed) as do the the matching newer, wxPython. Besides, Python 2 won't be maintained for ever. 
If it breaks for whatever reason down the road it might be trickier to fix if still based on an older version of Python.
That's my 2 [low denomination] of [favourite currency].

**Reply by BjornFJohansson** [reply]
I run on Python 3.8 and wxpython 4.1.1.

![image](https://user-images.githubusercontent.com/9726401/207610745-ca0cfe25-f275-4baa-abbe-11cd7a1d666b.png)

![image](https://user-images.githubusercontent.com/9726401/207611032-b4d06bcc-bb5c-4249-a57d-882bdf394bee.png)

Although I do some smaller software projects in Python I have never quite grasped how WikipPad works.
There are quite a few features that I wonder if people are still using. I never use the tree view and I never used any database but original sqlite for example.

Granted that there are unused features, perhaps a streamlined wikidpad could be made that is easier to understand and to maintain.

**Comment by clearoak** [comment]
Tree view is standard for me.  Get confused without it.   But I never use
the pretty view.  It's always in edit mode.  I'm noting, not making web
pages.

**Reply by BjornFJohansson** [reply]
Perhaps we all only use half of the features, but it is not the same half for everyone.

**Reply by safrye** [reply]
Thanks for keeping this program going!
I still use Wikidpad on a daily bases, at home at and at work. I always use the treeview and switch a lot between editing mode and preview mode as I have a lot of pictures (mostly screen captures of some program option screens and graphs from papers) integrated.
I installed your version under Mamba but don't use any plugins.
I am working under Linux and the only functions that I realised  don't work are jumping to the anchor position and pasting pictures. Would be nice to have that working. :) Unfortunately my Python skills are not even basic. :(

**Reply by BjornFJohansson** [reply]
I have since moved on to Obsidian. It does for me everything WP did and more and has a huge community. I recommend it.

**Reply by safrye** [reply]
Understandabel. :)
Do you know about an easy way to convert the data from WP to Obsidian?

P.S.: Found your post on Obsidians site (https://forum.obsidian.md/t/script-for-migrating-from-wikidpad/73307). I'll try it.

**Reply by benkinooby** [reply]
I have read and heard good things about Obisidian. I prefer the open source nature of WikidPad though and would rather invest some effort in keeping WikidPad alive (in a healthy way) than switching to Obsidian. That's just my opinion though, which I try to back up with code contributions.

**Reply by safrye** [reply]
I have now fully switched to Obsidian and probably would not make the switch back as I have added a lot of pages in the last time. I'm still cleaning up old pages that didn't convert perfectly but I don't mind as they are old pages.
Obsidian is not FOSS but the data is stored in single Markdown files which I can synchronise with Syncthing, something I didn't do with the  SQlite database in Wikidpad. With Synthing and MD files I only get sometimes single files that are out of sync but that does not destroy the whole data structure.
The data format is also the reason why I didn't switch to Joplin or Zettlr.

**Reply by ljwolfe** [reply]
I just haven't been able to get the hang of Obsidian -- every time I try, even keeping new-app-learning-curve in mind, it's usually less than half an hour before I run into something that sends me back to Wikidpad. One of the things that drives me nuts is missing the tree view, with lists of pages/tags/etc.

Do the folks who've made the switch have any advice, or suggestions where to look for advice? I have over a dozen wikidpad wikis with probably thousands of pages between them, so switching isn't going to be trivial under the best of circumstances.

I wish I'd kept up with coding after leaving college, but my 80s-era skills just aren't up to modern programming and I don't have the time I'd like to catch up so I can contribute to beloved projects like this one :-(.

**Reply by safrye** [reply]
I had the same problems when switching to Obsidian but my coding is even worse. I only do some bash scripting for my work and never found enough time to get into Python.
Obsidian has a lot of plugins but it's hard to find the right one for your needs. For the tree view I use "File Tree Alternative". The other community plugins I use are:
"Dataview", "Image Converter", "Link Tree", "Link with alias", and "Various Complements".
I also started using a folder structure in the vault to get a better overview but I'm still irritated by the way Obsidian handles attached files.
After one year there is probably no way back for me. The advantages are for me the easy synchronisation with Syncthing and that I could combine my wikis at home and two different work places just by dropping all in one folder.

**Comment by OpenSourceEverything** [comment]
FYI given that it sounds like WikidPad is dead here, I plan to try and maintain the master branch of my fork with updates to make it usable:

https://github.com/opensourceeverything/Wikidpad

**Reply by BjornFJohansson** [reply]
Good luck!

**Reply by benkinooby** [reply]
Hi @OpenSourceEverything thanks for picking up this issue. I tried to implement the bare minimum of updates to keep WikidPad functional with newer releases of python, wxpython and wxwidgets. I shared my work in the respective branches:

https://github.com/WikidPad/WikidPad/tree/python3.11_wxpython4.2.0_wxwidgets3.2

https://github.com/WikidPad/WikidPad/tree/python3.12.3_wxpython4.2.1_wxwidgets3.2.2.1

Those fixes allow WikidPad to start and run with the given set of python, wxpython and wxwidgets, but they ignore any runtime warnings and might break functionality that I did not test/use.

Your fork seems to spend much more deliberation on testing and CI. I welcome that and it's certainly something my branches are lacking. I will have a look at your fork and am inclined to switch to it - unless we get this WikidPad repository responsive again.

**Comment by benkinooby** [comment]
I am closing this discussion. The main question or concern of this discussion is answered in https://github.com/WikidPad/WikidPad/discussions/103 with the final comment https://github.com/WikidPad/WikidPad/discussions/103#discussioncomment-17537758
